docs/contributor/03-11-hap-rules.md

```diff
@@ -0,0 +1,249 @@
+    - gcp(HR=europe-west3)          # valid entry, new HR attribute makes the rule unique
+    - gcp(PR=*, HR=europe-west3)    # invalid rules, the same as previous one because of addition of `PR=*` attribute
+```
+
+Rule configuration must contain only unique entries in the scope of that list.
+If rule configuration contains duplicated rule entries an error that fails KEB startup is returned.
+
+Rule entry priority is selected by sorting all rule entries that apply to the request by the number of identification attributes they contain. 
+For example, a rule with no attributes and only a plan has lower priority then a rule with the same plan but a platform region attribute (`gcp` < `gcp(PR=cf-sa30)`).
+After sorting the entry that specifies most attributes (hence is the most specific) is selected. 
+Input attributes with value `*` are not taken into account when calculating priority.
+
+The following example shows priority of rules starting from the lowest priority:
+
+```
+aws -> S                              # search labels: hyperscalerType: aws, shared: true
+aws(PR=cf-eu11) -> EU                 # search labels: hyperscalerType: aws_cf-eu11, euAccess: true
+aws(PR=cf-eu11, HR=westeu) -> EU, S   # search labels: hyperscalerType: aws_cf-eu11_westeu, shared: true, euAccess: true
+```
+
+## Validation
+
+KEB validates HAP Rules during startup. If the configuration is invalid, KEB will not start and an error message will be displayed in the logs. 
+
+The constraints used for validation during KEB startup include:
+* Rules format check - all the rules must comply with the format specified above.
+* Every supported plan needs at least one rule entry, if no rule entry is defined for a plan then an error is returned during KEB startup.
+* Uniqueness validation check - KEB checks if all rule entries are unique in the scope of the rule. Specifying more than one entry with the same number of identification attributes is prohibited. If more than one such attribute is specified then the error is returned failing KEB startup. For more details see [Uniqnuess & Priority](#uniqnuess--priority) section. 
+
+
+## Initial Configuration
+
+The last example shows initial configuration created to mimic the current bahaviour of KEB at the time of writing the document. The configuration enforces that:
+* azure, aws, gcp have their own pools of dedicated bindings.
+* free plan uses aws or azure dedicated bindings depending on the provider value.
+* gcp clusters in the region cf-sa30 use the pool of secret bindings marked with labels: `hyperscalerType: gcp_cf-sa30`,
+* sap-converged-cloud clusters use the pool of secret bindings marked with labels: `hyperscalerType: openstack_<HYPERSCALER_REGION>` and all these pools are shared.
+* trial clusters can use one of two pools of shared secret bindings marked with labels: `hyperscalerType: azure` or `hyperscalerType: aws` depending on the used trial provider type.
```

Comment:
```suggestion
* Trial clusters can use one of two pools of shared secret bindings marked with labels `hyperscalerType: azure` or `hyperscalerType: aws`, depending on the trial provider type used.
```

Reply:
Trial is a plan name written in lower case on the same basis as in previous bullet points.